From f664d1d41d5c4beeac5b81e485c3498f13109db7 Mon Sep 17 00:00:00 2001
From: Rob Wu <rob@robwu.nl>
Date: Fri, 30 Mar 2018 16:33:52 +0200
Subject: [PATCH 24/24] Use correct index in process_notif_dev_connect

`devices[dev_idx]` should have been `devices[dev_idx - 1]`, aka `dev`.
---
 ltunify.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/ltunify.c
+++ b/ltunify.c
@@ -427,7 +427,7 @@ bool process_notif_dev_connect(struct hi
 	if (device_index) *device_index = dev_idx;
 	if (is_new_device) *is_new_device = !dev->device_present;
 
-	memset(&devices[dev_idx], 0, sizeof devices[dev_idx]);
+	memset(dev, 0, sizeof *dev);
 	dev->device_type = dcon->device_info & DEVCON_DEV_TYPE_MASK;
 	dev->wireless_pid = (dcon->pid_msb << 8) | dcon->pid_lsb;
 	dev->device_present = true;
